### PR TITLE
PN-10236-10237-10238-10239-10240-10241

### PR DESCRIPTION
**Pull request by Gabriiii99**
## Short description

Created new tests: 10236 - 10237 - 10238 - 10239 - 10240 - 10241 - 10259

## List of changes proposed in this pull request

- Created new file with notification cancellation tests. The notification is created and when it reaches a certain state is cancelled
- New general step to verify the notification state in the list of the main page (notification serached by position) in 2 minutes span
- Added some others steps where needed
- PN - 10259: added late, it checks if the destinatario see the notication cancellation

## How to test

Simply run `mvn clean test -D"cucumber.filter.tags=@annullamentoNotifica"` and verify if the tests works fine.

Run on AWS: https://eu-south-1.console.aws.amazon.com/codesuite/codebuild/151559006927/projects/pn-RunFeE2eTestsCodeBuildProject/build/pn-RunFeE2eTestsCodeBuildProject%3Acd58c55f-9efe-400b-840c-408316b44ac4?region=eu-south-1

- [x] Does this PR require a manual test in AWS CodeBuild?

### CODE REVIEW

**Comment by Gabriiii99** [comment]
Need to discuss the change about the method that control the notification corret creation

File: src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java [hunk 1]
```diff
@@ -37,22 +40,19 @@ public void waitLoadDettaglioNotificaDESection() {
...
-
-            getWebDriverWait(10).withMessage("il titolo Dettaglio notifica non è visibile").until(ExpectedConditions.visibilityOfElementLocated(titleDettaglioNotificaField));
-            getWebDriverWait(10).withMessage("il bottone indietro non è visibile").until(ExpectedConditions.visibilityOfElementLocated(indietroButtonBy));
-            getWebDriverWait(10).withMessage("Dettaglio notifica non è visibile").until(ExpectedConditions.visibilityOfElementLocated(informazioniBy));
-            getWebDriverWait(10).withMessage("La sezione Documenti allegati non è visibile").until(ExpectedConditions.visibilityOfElementLocated(allegatiSection));
-            getWebDriverWait(10).withMessage("Lo stato della notificanon non è visibile").until(ExpectedConditions.visibilityOfElementLocated(statoNotificaBy));
-            if (isSelfcare) {
-                getWebDriverWait(10).withMessage("Il sezione recapiti non è visibile").until(ExpectedConditions.visibilityOfElementLocated(sezioneRecapiti));
+            this.getWebDriverWait(10).withMessage("il titolo Dettaglio notifica non è visibile").until(ExpectedConditions.visibilityOfElementLocated(titleDettaglioNotificaField));
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java]: delete this

File: src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java [hunk 2]
```diff
@@ -819,6 +825,91 @@ public void verificaNotificaCreata() {
         }
     }
 
+    public void clickBottoneAnnullaNotifica() {
+        try {
+            By bottoneAnnullaNotifica = By.xpath("//button[@data-testid='cancelNotificationBtn']");
+            getWebDriverWait(10).until(ExpectedConditions.and(
+                    ExpectedConditions.visibilityOfElementLocated(bottoneAnnullaNotifica),
+                    ExpectedConditions.elementToBeClickable(bottoneAnnullaNotifica)));
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java]: delete /n

File: src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java [hunk 3]
```diff
@@ -819,6 +825,91 @@ public void verificaNotificaCreata() {
...
+
+        }
+    }
+
+    public void clickAnnullaNotificaModale() {
+        try {
+            By bottoneAnnullaNotificaModale = By.xpath("//button[@data-testid='modalCloseAndProceedBtnId']");
+            getWebDriverWait(10).until(ExpectedConditions.and(
+                    ExpectedConditions.visibilityOfElementLocated(bottoneAnnullaNotificaModale),
+                    ExpectedConditions.elementToBeClickable(bottoneAnnullaNotificaModale)));
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java]: as before

File: src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java [hunk 4]
```diff
@@ -819,6 +825,91 @@ public void verificaNotificaCreata() {
...
+
+    public void checkPopUpConfermaAnnullamentoNotifica() {
+        try {
+            By popUpConfermaAnnullamento = By.xpath("//div[@role='alert']/div[text()='La richiesta di annullamento è stata accettata.']");
+            getWebDriverWait(10).withMessage("Pop up NON visualizzato").until(ExpectedConditions.visibilityOfElementLocated(popUpConfermaAnnullamento));
+        } catch (TimeoutException e) {
+            logger.error("Il pop up di conferma dell'annullamento della notifica non viene visualizzato con errore: " + e.getMessage());
+            Assert.fail("Il pop up di conferma dell'annullamento della notifica non viene visualizzato con errore: " + e.getMessage());
+        }
+    }
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java]: format file

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RicercaNotifichePersonaFisicaPATest.java [hunk 5]
```diff
@@ -37,6 +37,7 @@ public void siVisualizzaCorrettamenteLaPaginaPiattaformaNotificheDestinatario()
         notifichePFPage.waitLoadNotificheDEPage();
     }
 
+    @And("Nella pagina Piattaforma Notifiche persona fisica inserire il codice IUN da dati notifica {string}")
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RicercaNotifichePersonaFisicaPATest.java]: present two AND, control it

File: src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java [hunk 6]
```diff
@@ -70,7 +70,7 @@ public void clickLinkAttestazioniOpponibile(int numeroLinkAttestazioniOpponibile
         if (attestazioniFile.get(numeroLinkAttestazioniOpponibile).isDisplayed()) {
             attestazioniFile.get(numeroLinkAttestazioniOpponibile).click();
         } else {
-            this.js().executeScript("arguments[0].scrollIntoView(true);", attestazioniFile.get(numeroLinkAttestazioniOpponibile));
+            js().executeScript("arguments[0].scrollIntoView(true);", attestazioniFile.get(numeroLinkAttestazioniOpponibile));
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java]: Maybe you can actually use this:
https://github.com/pagopa/pn-frontend-e2e/blob/5b46832c7ab76e1ca9477cccf1c933698c363b9a/src/main/java/it/pn/frontend/e2e/common/BasePage.java#L46-L57

File: src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java [hunk 7]
```diff
@@ -79,7 +79,7 @@ public void clickLinkDocumentiAllegati(int numeroLinkDocumentiAllegati) {
         if (documentiAllegati.get(numeroLinkDocumentiAllegati).isDisplayed()) {
             documentiAllegati.get(numeroLinkDocumentiAllegati).click();
         } else {
-            this.js().executeScript("arguments[0].scrollIntoView(true);", attestazioniFile.get(numeroLinkDocumentiAllegati));
+            js().executeScript("arguments[0].scrollIntoView(true);", attestazioniFile.get(numeroLinkDocumentiAllegati));
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java]: Same for this one:
https://github.com/pagopa/pn-frontend-e2e/blob/5b46832c7ab76e1ca9477cccf1c933698c363b9a/src/main/java/it/pn/frontend/e2e/common/BasePage.java#L46-L57

**Review by afus0 — CHANGES_REQUESTED** [review]

File: src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java (hunk 7 above)
Comment (Gabriiii99) [reply src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java]: This method has no usage in the project

File: src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java (hunk 6 above)
Comment (Gabriiii99) [reply src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java]: this is an old method, a lot of tests that use its don't work

**Review by afus0 — APPROVED** [review]

File: src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java [hunk 8]
```diff
@@ -796,19 +791,51 @@ public void verificaNotificaCreata() {
...
                 maximumRetry++;
             }
         } while (notificationStatus.equals("WAITING"));
         driver.navigate().refresh();
         logger.info("La notifica è stata creata correttamente");
     }
 
+
+    public void checkStatoNotifica(String stato) {
+        try {
```
Comment (azhansultan) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java]: used try

Comment (leleOFA) [reply src/main/java/it/pn/frontend/e2e/pages/mittente/PiattaformaNotifichePage.java]: done

**Review by leleOFA — APPROVED** [review]

**Pull request merged**